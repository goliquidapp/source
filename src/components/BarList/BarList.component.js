import React, {Component} from 'react';
import {ScrollView, Text, View, ActivityIndicator, Dimensions} from 'react-native';
import * as Animatable from 'react-native-animatable';

class BarList extends Component{
	render(){
		return (
				<Animatable.View animation={"fadeIn"} useNativeDriver duration={800} style={[styles.container,this.props.containerStyle]}>
					<ScrollView ref={(ref)=>this.scrollView=ref}
								onScrollBeginDrag={this.handleScrollBegin} 
								onMomentumScrollEnd={this.handleScrollEnd} 
								showsHorizontalScrollIndicator={false} 
								directionalLockEnabled={true} 
								horizontal={true} 
								pagingEnabled={true}
								snapToInterval={this.props.elementSize} 
								scrollEnabled
								contentContainerStyle={styles.scrollContainer}>
						{this.props.children}
					</ScrollView>
				</Animatable.View>
			)
	}
}

const styles={
	container:{
		position:'relative'
	},
	scrollContainer:{
		flexDirection:'row', 
		alignItems:'center'
	}
}


export default BarList