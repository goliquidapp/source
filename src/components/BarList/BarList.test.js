import React from 'react';
import { Text, View} from 'react-native';
import renderer from 'react-test-renderer';

import BarList from './BarList.component';

describe("Testing BarList component",()=>{
	it('renders correctly', () => {
		const wrapper = renderer.create(
    		<BarList>
    			<Text>Text 1</Text>
    			<View>
    				<Text>View 1</Text>
    			</View>
    			<View>
    				<Text>View 2</Text>
    			</View>
    			<View>
    				<Text>View 3</Text>
    			</View>
    			<Text>Text 2</Text>
    		</BarList>
    	).toJSON();
		expect(wrapper).toMatchSnapshot();
	});
})