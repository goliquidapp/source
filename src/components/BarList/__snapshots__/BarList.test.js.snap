// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing BarList component renders correctly 1`] = `
<View
  style={
    Object {
      "opacity": 0,
      "position": "relative",
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
    directionalLockEnabled={true}
    horizontal={true}
    pagingEnabled={true}
    scrollEnabled={true}
    showsHorizontalScrollIndicator={false}
  >
    <View>
      <Text>
        Text 1
      </Text>
      <View>
        <Text>
          View 1
        </Text>
      </View>
      <View>
        <Text>
          View 2
        </Text>
      </View>
      <View>
        <Text>
          View 3
        </Text>
      </View>
      <Text>
        Text 2
      </Text>
    </View>
  </RCTScrollView>
</View>
`;
